src/laser_geometry.cpp

```diff
@@ -437,7 +437,8 @@ const boost::numeric::ublas::matrix<double>& LaserProjection::getUnitVectors_(do
     {
       //check to see if we want to keep the point
       const float range = scan_in.ranges[i];
-      if ((preservative && std::isfinite(ranges(i,0))) || (range < range_cutoff && range >= scan_in.range_min))
```

Comment:
Or set everything below ```range_min``` to ```range_min``` as well in the check on line 301. 

Also the check on 440 as you have it now will keep the NAN even in non `preservative` mode. If you crop values to ```range_min``` the following would be enough:

`if (preservative || (range < range_cutoff && range >= scan_in.range_min))`